src/main/scala/java/util/logging/Logger.scala

```diff
@@ -6,38 +6,68 @@ import scala.collection.mutable
 object Logger {
 
   val GLOBAL_LOGGER_NAME: String =  "global"
+  private val ROOT_LOGGER_NAME: String = ""
 
   // Not implemented, deprecated on JDK 1.8
   //val global: Logger
 
   private val defaultLogLevel: Level = Level.INFO
 
-  private val loggers: mutable.Map[String, Logger] = mutable.Map.empty
+  private lazy val loggers: mutable.Map[String, Logger] = mutable.Map.empty
+
+  /**
+    * Find parent logger
+    */
+  @tailrec
+  private def findParentLoggerOf(name:String): Logger = {
+    name match {
+      case null => // Anonymous logger
+        rootLogger
+      case ROOT_LOGGER_NAME =>
+        null
+      case _ =>
+        val parentNodes = name.split("\\.").toList.dropRight(1)
+        val parentName = parentNodes.mkString(".")
+        loggers.get(parentName) match {
+          case Some(l) => l
+          case None =>
+            findParentLoggerOf(parentName)
+        }
+    }
+  }
 
   private def newLogger(name: String): Logger = {
     val logger = new Logger(name, null)
     logger.setLevel(null)
     logger.setUseParentHandlers(true)
-    logger.setParent(rootLogger)
+    logger.setParent(findParentLoggerOf(name))
+    if (name != null) {
+      updateChildLoggerParent(logger)
+    }
     logger
   }
 
+  private def updateChildLoggerParent(newParent:Logger): Unit ={
+    val prefix = s"${newParent.getName}."
+    for ((name, l) <- loggers if name.startsWith(prefix)) {
+      val currentParent = l.getParent
+      // For example, if a new parent is a.b and the child is a.b.c.d,
+      // unless the current parent is a.b.c, we need to update the parent
+      if (currentParent == null || !currentParent.getName.startsWith(prefix)) {
+        l.setParent(newParent)
+      }
```

Comment:
I'm confused here. I don't really understand the comment, and I also don't understand how the condition of the `if` reflects what is described by the comment. Could you elaborate, please?